R/FilterStateRange.R

```diff
@@ -214,6 +214,117 @@ RangeFilterState <- R6::R6Class( # nolint
     #'
     set_selected = function(value) {
       super$set_selected(value)
+    },
+
+    #' @description
+    #' Shiny module server.
+    #'
+    #' @param id (`character(1)`)\cr
+    #'   shiny module instance id
+    #'
+    #' @return `moduleServer` function which returns reactive value
+    #'   signaling that remove button has been clicked
+    #'
+    server = function(id) {
+      moduleServer(
+        id = id,
+        function(input, output, session) {
+
+          observeEvent(self$get_keep_na(), {
+            
+            if (self$get_keep_na()) {
+                class <- "fa fa-check"
+              } else {
+                class <- "fa fa-xmark"
+              }
+
+            output$header_keep_na <- renderUI({
+              tagList(
+                tags$strong("NA "),
+                tags$span(class = class)
+              )
+            })
+          })
+
+          observeEvent(self$get_keep_inf(), {
+
+            if (is.null(self$get_keep_inf())) {
+              inf_tag <- tags$span()
+            } else {
+              inf_tag <- tagList(
+                tags$strong("Inf "),
+                tags$span(
+                  class = if (self$get_keep_inf()) "fa fa-check" else "fa fa-xmark"
+                )
+              )
+            }
+
+            output$header_keep_inf <- renderUI({
+              inf_tag
+            })
+          })
+
+          observeEvent(self$get_selected(), {
+            if (length(self$get_selected() > 1)) {
+              value <- paste0(
+                "(",
+                paste(self$get_selected(), collapse = "-"),
+                ")"
+              )
+            } else {
+              value <- self$get_selected()
+            }
+
+            output$header_name_value <- renderUI({
+              tagList(
+                tags$strong(self$get_varname(deparse = TRUE)),
+                tags$span(value)
+              )
+            })
+          })
+          private$server_inputs("inputs")
+          reactive(input$remove) # back to parent to remove self
+        }
+      )
+    },
+
+    #' @description
+    #' Shiny module UI.
+    #'
+    #' @param id (`character(1)`)\cr
+    #'  shiny element (module instance) id;
+    #'  the UI for this class contains simple message stating that it is not supported
+    #'
+    ui = function(id) {
+      ns <- NS(id)
+      
+      tags$li(
+        id = id,
+        tags$div(
+        class = "filter-card",
+        tags$div(
+          class = "filter-card-header",
+          uiOutput(ns("header_name_value"), inline = TRUE),
+          uiOutput(ns("header_keep_na"), inline = TRUE),
+          uiOutput(ns("header_keep_inf"), inline = TRUE),
```

Comment:
Maybe instead of: 
`<varname>: min-max NA: TRUE Inf: TRUE`
`<varname>: min-max NA: FALSE Inf: FALSE`
 
following could be better:
`<varname>: min-max, NA, Inf`
`<varname>: min-max`

Reply:
Not sure what you mean here. Could you expand on this?

Reply:
`NA: TRUE` and `Inf: TRUE` takes some space so I think about alternative display. Instead of having `NA: TRUE` as a separate "span" we can include it in the choices, so we could have:
- `SEX: F, M, NA` when NA is TRUE
- `AGE: 18-64, Inf` when Inf is TRUE
- `AGE: 18-64` when Inf is FALSE